internal/services/cdn/frontdoorruleactions/cdn_frontdoor_rule_actions.go

```diff
@@ -400,41 +405,41 @@ func FlattenCdnFrontDoorUrlRewriteAction(input cdn.URLRewriteAction) map[string]
 	}
 }
 
-func FlattenCdnFrontDoorRouteConfigurationOverrideAction(input cdn.DeliveryRuleRouteConfigurationOverrideAction) (map[string]interface{}, error) {
-	queryStringCachingBehavior := ""
-	cacheBehavior := ""
-	compressionEnabled := false
-	cacheDuration := ""
+func FlattenCdnFrontDoorRouteConfigurationOverrideAction(input rules.DeliveryRuleRouteConfigurationOverrideAction) (map[string]interface{}, error) {
+	var queryStringCachingBehavior string
+	var cacheBehavior string
+	var compressionEnabled bool
+	var cacheDuration string
+	var forwardingProtocol string
+	var originGroupId string
 	queryParameters := make([]interface{}, 0)
-	forwardingProtocol := ""
-	originGroupId := ""
-
-	if params := input.Parameters; params != nil {
-		if config := params.CacheConfiguration; config != nil {
-			queryStringCachingBehavior = string(config.QueryStringCachingBehavior)
-			cacheBehavior = string(config.CacheBehavior)
-			compressionEnabled = config.IsCompressionEnabled == cdn.RuleIsCompressionEnabledEnabled
-			queryParameters = flattenCsvToStringSlice(config.QueryParameters)
-
-			if config.CacheDuration != nil {
-				cacheDuration = *config.CacheDuration
-			}
-		} else {
-			cacheBehavior = string(cdn.RuleIsCompressionEnabledDisabled)
-		}
 
-		if override := params.OriginGroupOverride; override != nil {
-			forwardingProtocol = string(override.ForwardingProtocol)
+	params := input.Parameters
 
-			// NOTE: Need to normalize this ID here because if you modified this in portal the resourceGroup comes back as resourcegroup.
-			// ignore the error here since it was set on the resource in Azure and we know it is valid.
-			originGroup, err := parse.FrontDoorOriginGroupIDInsensitively(*override.OriginGroup.ID)
-			if err != nil {
-				return nil, err
-			}
+	if config := params.CacheConfiguration; config != nil {
+		queryStringCachingBehavior = string(pointer.From(config.QueryStringCachingBehavior))
+		cacheBehavior = string(pointer.From(config.CacheBehavior))
+		compressionEnabled = pointer.From(config.IsCompressionEnabled) == rules.RuleIsCompressionEnabledEnabled
+		queryParameters = flattenCsvToStringSlice(config.QueryParameters)
 
-			originGroupId = originGroup.ID()
+		if config.CacheDuration != nil {
+			cacheDuration = *config.CacheDuration
 		}
+	} else {
+		cacheBehavior = string(rules.RuleIsCompressionEnabledDisabled)
+	}
+
+	if override := params.OriginGroupOverride; override != nil {
+		forwardingProtocol = string(pointer.From(override.ForwardingProtocol))
+
+		// NOTE: Need to normalize this ID here because if you modified this in portal the resourceGroup comes back as resourcegroup.
+		// ignore the error here since it was set on the resource in Azure and we know it is valid.
+		originGroup, err := origins.ParseOriginIDInsensitively(*override.OriginGroup.Id)
```

Comment:
Is this the correct ID type here? origin vs originGroup?

Reply:
DOH! No that is not, totally missed that... 🙉 Fixed.